File: MB-Secretaria/confidence_py/evidence_evaluator.py
from .types import ClinicalConfidenceInput, EvidenceEvaluation, LayerSummary


def _score_layer(layer: LayerSummary | None) -> float:
    if layer is None or not layer.available:
        return 0.0
    return max(0.0, min(1.0, layer.quality_score if layer.quality_score is not None else 0.6))


def evaluate_evidence(item: ClinicalConfidenceInput) -> EvidenceEvaluation:
    missing: list[str] = []
    conflicts: list[str] = []
    image_needed = item.modality in {"image", "multimodal"}

    layers = [
        ("retrieval_unavailable", item.retrieval_summary),
        ("memory_context_unavailable", item.memory_context_summary),
        ("risk_summary_unavailable", item.risk_summary),
    ]
    if image_needed:
        layers.append(("image_summary_unavailable", item.image_summary))

    for missing_name, layer in layers:
        if layer is None or not layer.available:
            missing.append(missing_name)
        conflicts.extend(layer.conflicts if layer else [])

    retrieval = _score_layer(item.retrieval_summary)
    memory = _score_layer(item.memory_context_summary)
    risk = _score_layer(item.risk_summary)
    image = _score_layer(item.image_summary) if image_needed else 1.0

    return EvidenceEvaluation(
        evidence_completeness=round((retrieval + memory + risk + image) / 4, 3),
        missing_evidence=missing,
        conflict_signals=conflicts,
    )

File: MB-Secretaria/confidence/evidence-evaluator.ts
import type { ClinicalConfidenceInput, EvidenceEvaluation } from "./types";

function scoreLayer(available: boolean, quality?: number): number {
  if (!available) return 0;
  return Math.max(0, Math.min(1, quality ?? 0.6));
}

export function evaluateEvidence(input: ClinicalConfidenceInput): EvidenceEvaluation {
  const missing: string[] = [];
  const conflicts: string[] = [];
  const retrieval = scoreLayer(input.retrieval_summary.available, input.retrieval_summary.quality_score);
  const memory = scoreLayer(input.memory_context_summary.available, input.memory_context_summary.quality_score);
  const risk = scoreLayer(input.risk_summary.available, input.risk_summary.quality_score);
  const imageNeeded = input.modality === "image" || input.modality === "multimodal";
  const image = imageNeeded ? scoreLayer(Boolean(input.image_summary?.available), input.image_summary?.quality_score) : 1;

  if (!input.retrieval_summary.available) missing.push("retrieval_unavailable");
  if (!input.memory_context_summary.available) missing.push("memory_context_unavailable");
  if (!input.risk_summary.available) missing.push("risk_summary_unavailable");
  if (imageNeeded && !input.image_summary?.available) missing.push("image_summary_unavailable");

  for (const layer of [input.retrieval_summary, input.memory_context_summary, input.risk_summary, input.image_summary]) {
    conflicts.push(...(layer?.conflicts ?? []));
  }

  return {
    evidence_completeness: Number(((retrieval + memory + risk + image) / 4).toFixed(3)),
    missing_evidence: missing,
    conflict_signals: conflicts,
  };
}
